gammapy/scripts/image_pipe.py

```diff
@@ -21,37 +28,152 @@ def image_pipe_main(args=None):
 
 class ImageAnalysis(object):
     """Gammapy 2D image based analysis.
+
+    center : `astropy.coordinates.SkyCoord`
+        Center of the image
+    energy_band : `~gammapy.utils.energy.Energy
+        Energy band for which we want to compute the image
+    offset_band :`astropy.coordinates.Angle`
+        Offset Band where you compute the image
+    data_store : `~gammapy.data.DataStore`
+        `DataStore` where are situated the events
     """
 
-    def __init__(self, config):
-        self.config = config
-
-    @classmethod
-    def from_yaml(cls, filename):
-        """Read config from YAML file."""
-        import yaml
-        log.info('Reading {}'.format(filename))
-        with open(filename) as fh:
-            config = yaml.safe_load(fh)
-        return cls(config)
-
-    def run(self):
-        """Run analysis chain."""
-        log.info('Running analysis ...')
-        print(self.config['general']['outdir'])
-        print(self.config)
-
-    def make_counts(self):
+    def __init__(self, center, energy_band, offset_band, data_store, counts_image=None, bkg_image=None):
+        self.data_store = data_store
```

Comment:
This is an important question, how we handle observations and observation lists in algorithms.
I don't know yet what good patterns are.
LSST is passing around `Butler` objects (the `DataStore` equivalent) in high-level scripts, and then individual pieces in functions.

For now my suggestion would be that we allow to handle this differently how people like (whatever works), and then at the Gammapy coding spring discuss the patterns that have emerged and which one (or few) we want to adopt throughout.